src/main/java/edu/ie3/datamodel/io/factory/input/Transformer2WInputFactory.java

```diff
@@ -37,6 +37,13 @@ protected Transformer2WInput buildModel(
       NodeInput nodeB,
       OperatorInput operator,
       OperationTime operationTime) {
+
+    if (nodeB.getVoltLvl().getNominalVoltage().compareTo(nodeA.getVoltLvl().getNominalVoltage())
+        > 0) {
+      throw new IllegalArgumentException(
+          "nodeA must be on the upper voltage side of the transformer");
+    }
```

Comment:
I would move that into the actual constructor of the class `Transformer2WInput`. Since there are different constructors make sure that the method gets called in each of them.